src/wallet/coin_selection.rs

```diff
@@ -408,10 +472,22 @@ impl<D: Database> CoinSelectionAlgorithm<D> for BranchAndBoundCoinSelection {
             .expect("Bitcoin amount to fit into i64");
 
         if curr_value > actual_target {
+            let selected_amount = required_utxos
+                .iter()
+                .map(|u| u.effective_value)
+                .sum::<i64>();
```

Comment:
Isn't this literally `curr_value`? Why do you have to recompute it here?

Reply:
So true!